lightning/src/routing/router.rs

```diff
@@ -181,7 +215,20 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
 		return Err(LightningError{err: "Cannot generate a route of more value than all existing satoshis".to_owned(), action: ErrorAction::IgnoreError});
 	}
 
-	// We do a dest-to-source Dijkstra's sorting by each node's distance from the destination
+	// The general route finding idea is the following:
+	// 1. Fill first/last hops communicated by the caller.
+	// 2. Construct and store a path.
+	//	Remember which channels were used so that future paths don't rely on the same capacity.
+	//	Remember the used channel with minimum capacity ("bottleneck"), so that future paths avoid using it.
+	// 3. If we hit the recommended target amount or made too many attempts of path finding, proceeed to the next step. Otherwise, repeat step 2.
+	// 4. At this point, the found paths may be using higher amount than we need to send.
+	// 5. Of all the found paths, select only those with the lowest total fee.
+	// 6. Randomly combine paths into routes having enough to fulfill the payment.
```

Comment:
If path weight is the fee used by each, a knapsack based on their capacity would too CPU intensive ?

Reply:
Depends on the set size :)
I think this optimization should be done as a follow-up. Maybe some constrained knapsack would work.